test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6362,3 +6415,774 @@ def test_release_nodes_shared_server_restart_immed(self):
+        except PtlException as e:
+            self.assertTrue("illegal -W value" in e.msg[0])
+
+        a = {ATTR_W: "release_nodes_on_stageout=tru"}
+        j = Job(TEST_USER, a)
+        try:
+            jid = self.server.submit(j)
+        except PtlException as e:
+            self.assertTrue("illegal -W value" in e.msg[0])
+
+    def test_resc_accumulation(self):
+        """
+        Test that resources gets accumulated when a mom is released
+        """
+
+        # skip this test due to PP-972
+        self.skip_test(reason="Test fails due to PP-972")
+
+        # Create custom resources
+        attr = {}
+        attr['type'] = 'float'
+        attr['flag'] = 'nh'
+        r = 'foo_f'
+        self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+
+        attr1 = {}
+        attr1['type'] = 'size'
+        attr1['flag'] = 'nh'
+        r1 = 'foo_i'
+        self.server.manager(
+            MGR_CMD_CREATE, RSC, attr1, id=r1, runas=ROOT_USER, logerr=False)
+
+        self.logger.info("test_epilogue")
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executed epilogue hook")
+if e.job.in_ms_mom():
+    e.job.resources_used["vmem"] = pbs.size("9gb")
+    e.job.resources_used["foo_i"] = pbs.size(999)
+    e.job.resources_used["foo_f"] = 0.09
+else:
+    e.job.resources_used["vmem"] = pbs.size("10gb")
+    e.job.resources_used["foo_i"] = pbs.size(1000)
+    e.job.resources_used["foo_f"] = 0.10
+"""
+
+        hook_name = "epi"
+        a = {'event': "execjob_epilogue", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        jid = self.create_and_submit_job('job1_5')
+        self.server.expect(JOB, {'job_state': "R"}, id=jid)
+
+        # Release hostC
+        cmd = [self.pbs_release_nodes_cmd, '-j', jid, self.n7]
+        ret = self.server.du.run_cmd(self.server.hostname, cmd)
+        self.assertEqual(ret['rc'], 0)
+
+        self.momC.log_match("executed epilogue hook", max_attempts=10)
+        self.momC.log_match("DELETE_JOB2 received", max_attempts=10)
+
+        self.server.delete(jid, wait=True)
+
+        self.server.expect(JOB, {'job_state': 'F',
+                                 "resources_used.foo_i": "3kb",
+                                 "resources_used.foo_f": '0.29',
+                                 "resources_used.vmem": '29gb'}, id=jid)
+
+    @timeout(500)
+    def test_release_reservations(self):
+        """
+        Release nodes from a reservation wil throw error. However
+        jobs inside reservation queue works as expected.
+        """
+
+        # Create a reservation on multiple nodes
+        start = int(time.time()) + 60
+        a = {'Resource_List.select': self.job1_select,
+             'Resource_List.place': 'scatter',
+             'reserve_start': start}
```

Comment:
Do you not have to submit a duration or an end time?

Reply:
PTL will take default.